cpp/src/arrow/compute/row/grouper.cc

```diff
@@ -419,6 +414,14 @@ struct GrouperImpl : public Grouper {
     return std::move(impl);
   }
 
+  Status Reset() override {
+    map_.clear();
+    offsets_.clear();
+    key_bytes_.clear();
+    num_groups_ = 0;
```

Comment:
What about `encoders_`?

Reply:
From what I saw in https://github.com/apache/arrow/blob/main/cpp/src/arrow/compute/kernels/row_encoder_internal.h, none of the encoders holds states that would change during the lifespan of the grouper.